pkg/virt-controller/services/template.go

```diff
@@ -527,6 +498,43 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 	return &pod, nil
 }
 
+func VMICustomDebugFilters(vmi *v1.VirtualMachineInstance) string {
+	if customDebugFilters, exists := vmi.Annotations[v1.CustomLibvirtLogFiltersAnnotation]; exists {
+		return customDebugFilters
+	}
+	return ""
+}
+
+func (t *templateService) newVirtLauncherCommandRenderer(vmi *v1.VirtualMachineInstance, domainName string, gracePeriodSeconds int64, requestedHookSidecarList hooks.HookSidecarList) *VirtLauncherCommandRenderer {
+	var options []VirtLauncherCommandRendererOption
+	if util.IsNonRootVMI(vmi) {
+		options = append(options, WithNonRootUser())
+	}
+	if customDebugFilters := VMICustomDebugFilters(vmi); customDebugFilters != "" {
+		log.Log.Object(vmi).Infof("Applying custom debug filters for vmi %s: %s", vmi.Name, customDebugFilters)
+		options = append(options, WithLibvirtCustomDebugFilters(customDebugFilters))
+	}
+	return NewVirtLauncherCommandRenderer(
+		string(vmi.UID),
+		domainName,
+		int(gracePeriodSeconds),
+		len(requestedHookSidecarList),
+		vmi.GetNamespace(),
+		t.virtLauncherStaticConfig(),
+		append(options, t.launcherCommandRendererOptions(vmi)...)...,
+	)
```

Comment:
The API looks a bit too complex to me, and is probably prone to growing even bigger in the future...

Reply:
Can you suggest an alternative ? The launcher command receives a bunch of unrelated things; I haven't managed to aggregate this into anything were any sense of meaning could be inferred from.